adapters/repos/db/queue/queue.go

```diff
@@ -0,0 +1,1063 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package queue
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	// defaultChunkSize is the maximum size of each chunk file. Set to 100MB.
+	defaultChunkSize = 100 * 1024 * 1024
+
+	// chunkFileFmt is the format string for the chunk files,
+	chunkFileFmt = "chunk-%d.bin"
+
+	magicHeader = "WV8Q"
+)
+
+// regex pattern for the chunk files
+var chunkFilePattern = regexp.MustCompile(`chunk-\d+\.bin`)
+
+type Queue interface {
+	ID() string
+	Size() int64
+	DequeueBatch() (batch *Batch, err error)
+	Metrics() *Metrics
+}
+
+type BeforeScheduleHook interface {
+	BeforeSchedule() bool
+}
+
+var _ Queue = &DiskQueue{}
+
+type DiskQueue struct {
+	// Logger for the queue. Wrappers of this queue should use this logger.
+	Logger           logrus.FieldLogger
```

Comment:
yes so it can be used by types that embed the DiskQueue, this is what the `VectorIndexQueue` does for example.
That prevents them from creating their own logger.